In [1]:
import pandas as pd

import warnings
warnings.filterwarnings('ignore')
import itertools

from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"

In [2]:
df = pd.read_csv('credit_train.csv')

In [3]:
df.columns

Index(['ID', 'Customer_ID', 'Month', 'Name', 'Age', 'SSN', 'Occupation',
       'Annual_Income', 'Monthly_Inhand_Salary', 'Num_Bank_Accounts',
       'Num_Credit_Card', 'Interest_Rate', 'Num_of_Loan', 'Type_of_Loan',
       'Delay_from_due_date', 'Num_of_Delayed_Payment', 'Changed_Credit_Limit',
       'Num_Credit_Inquiries', 'Credit_Mix', 'Outstanding_Debt',
       'Credit_Utilization_Ratio', 'Credit_History_Age',
       'Payment_of_Min_Amount', 'Total_EMI_per_month',
       'Amount_invested_monthly', 'Payment_Behaviour', 'Monthly_Balance',
       'Credit_Score'],
      dtype='object')

# Preprocessing

In [4]:
df.head()

ID Customer_ID     Month           Name   Age          SSN Occupation  \
0  0x1602   CUS_0xd40   January  Aaron Maashoh    23  821-00-0265  Scientist   
1  0x1603   CUS_0xd40  February  Aaron Maashoh    23  821-00-0265  Scientist   
2  0x1604   CUS_0xd40     March  Aaron Maashoh  -500  821-00-0265  Scientist   
3  0x1605   CUS_0xd40     April  Aaron Maashoh    23  821-00-0265  Scientist   
4  0x1606   CUS_0xd40       May  Aaron Maashoh    23  821-00-0265  Scientist   

  Annual_Income  Monthly_Inhand_Salary  Num_Bank_Accounts  ...  Credit_Mix  \
0      19114.12            1824.843333                  3  ...           _   
1      19114.12                    NaN                  3  ...        Good   
2      19114.12                    NaN                  3  ...        Good   
3      19114.12                    NaN                  3  ...        Good   
4      19114.12            1824.843333                  3  ...        Good   

   Outstanding_Debt Credit_Utilization_Ratio     Credit_History_Age  \
0            809.98                26.822620  22 Years and 1 Months   
1            809.98                31.944960                    NaN   
2            809.98                28.609352  22 Years and 3 Months   
3            809.98                31.377862  22 Years and 4 Months   
4            809.98                24.797347  22 Years and 5 Months   

   Payment_of_Min_Amount Total_EMI_per_month Amount_invested_monthly  \
0                     No           49.574949       80.41529543900253   
1                     No           49.574949      118.28022162236736   
2                     No           49.574949         81.699521264648   
3                     No           49.574949       199.4580743910713   
4                     No           49.574949      41.420153086217326   

                  Payment_Behaviour     Monthly_Balance Credit_Score  
0   High_spent_Small_value_payments  312.49408867943663         Good  
1    Low_spent_Large_value_payments  284.62916249607184         Good  
2   Low_spent_Medium_value_payments   331.2098628537912         Good  
3    Low_spent_Small_value_payments  223.45130972736786         Good  
4  High_spent_Medium_value_payments  341.48923103222177         Good  

[5 rows x 28 columns]

### PreProcessing ID and Customer_ID

In [5]:
df_id = df.copy()

In [6]:
df_id['ID'].nunique()

100000

##### As ID is unique for all values. We can drop them.

In [7]:
df_id.drop(columns = 'ID', axis = 1, inplace = True)

In [8]:
df_id.head()

Customer_ID     Month           Name   Age          SSN Occupation  \
0   CUS_0xd40   January  Aaron Maashoh    23  821-00-0265  Scientist   
1   CUS_0xd40  February  Aaron Maashoh    23  821-00-0265  Scientist   
2   CUS_0xd40     March  Aaron Maashoh  -500  821-00-0265  Scientist   
3   CUS_0xd40     April  Aaron Maashoh    23  821-00-0265  Scientist   
4   CUS_0xd40       May  Aaron Maashoh    23  821-00-0265  Scientist   

  Annual_Income  Monthly_Inhand_Salary  Num_Bank_Accounts  Num_Credit_Card  \
0      19114.12            1824.843333                  3                4   
1      19114.12                    NaN                  3                4   
2      19114.12                    NaN                  3                4   
3      19114.12                    NaN                  3                4   
4      19114.12            1824.843333                  3                4   

   ...  Credit_Mix Outstanding_Debt Credit_Utilization_Ratio  \
0  ...           _           809.98                26.822620   
1  ...        Good           809.98                31.944960   
2  ...        Good           809.98                28.609352   
3  ...        Good           809.98                31.377862   
4  ...        Good           809.98                24.797347   

      Credit_History_Age Payment_of_Min_Amount Total_EMI_per_month  \
0  22 Years and 1 Months                    No           49.574949   
1                    NaN                    No           49.574949   
2  22 Years and 3 Months                    No           49.574949   
3  22 Years and 4 Months                    No           49.574949   
4  22 Years and 5 Months                    No           49.574949   

   Amount_invested_monthly                 Payment_Behaviour  \
0        80.41529543900253   High_spent_Small_value_payments   
1       118.28022162236736    Low_spent_Large_value_payments   
2          81.699521264648   Low_spent_Medium_value_payments   
3        199.4580743910713    Low_spent_Small_value_payments   
4       41.420153086217326  High_spent_Medium_value_payments   

      Monthly_Balance  Credit_Score  
0  312.49408867943663          Good  
1  284.62916249607184          Good  
2   331.2098628537912          Good  
3  223.45130972736786          Good  
4  341.48923103222177          Good  

[5 rows x 27 columns]

Customer_ID, Name, SSN all seems to be same

In [9]:
df_id['Customer_ID'].nunique()
df_id["SSN"].nunique()

12500

12501

Check How many unique SSN each customer_ID has

In [10]:
df_id.groupby(['Customer_ID', 'SSN'])['SSN'].unique()

Customer_ID  SSN        
CUS_0x1000   913-74-1218    [913-74-1218]
CUS_0x1009   #F%$D@*&8        [#F%$D@*&8]
             063-67-6938    [063-67-6938]
CUS_0x100b   #F%$D@*&8        [#F%$D@*&8]
             238-62-0395    [238-62-0395]
                                ...      
CUS_0xff3    726-35-5322    [726-35-5322]
CUS_0xff4    655-05-7666    [655-05-7666]
CUS_0xff6    541-92-8371    [541-92-8371]
CUS_0xffc    226-86-7294    [226-86-7294]
CUS_0xffd    832-88-8320    [832-88-8320]
Name: SSN, Length: 17115, dtype: object

In [11]:
unique_values = list(df_id[df_id['Customer_ID'] == 'CUS_0xd40']['SSN'].unique())
unique_values

['821-00-0265', '#F%$D@*&8']

Each Customer_ID seems to have a unique SSN except for some noisy data (#F%$D@*&8)

### PreProcessing SSN

In [24]:
df_ssn = df_id.copy()

In [13]:
(df_ssn["SSN"] == '#F%$D@*&8').sum()

5572

In [14]:
corresponding_unique_ssn_values = {}

In [15]:
unique_customer_id = list(df_ssn['Customer_ID'].unique())

In [16]:
list(df_ssn[df_ssn['Customer_ID'] == 'CUS_0xd40']['SSN'].unique())

['821-00-0265', '#F%$D@*&8']

Replace SSN according to corresponding Customer_ID

In [17]:
for id in unique_customer_id:
    corresponding_unique_ssn_values[id] = list(df_ssn[df_ssn['Customer_ID'] == id]['SSN'].unique())

In [18]:
dict(itertools.islice(corresponding_unique_ssn_values.items(), 5))

{'CUS_0xd40': ['821-00-0265', '#F%$D@*&8'],
 'CUS_0x21b1': ['004-07-5839'],
 'CUS_0x2dbc': ['486-85-3974'],
 'CUS_0xb891': ['072-31-6145', '#F%$D@*&8'],
 'CUS_0x1cdb': ['615-06-7821']}

In [19]:
(df_ssn['SSN'] == '#F%$D@*&8').sum()

count = 0
for i in range(len(df_ssn)):
    if(df_ssn['SSN'][i].startswith('#')):
        count += 1 
count

5572

5572

In [23]:
[val for val in corresponding_unique_ssn_values[df_ssn['Customer_ID'][0]] if not val.startswith('#')]
[val for val in corresponding_unique_ssn_values['CUS_0xd40'] if not val.startswith('#')]

'821-00-0265'

['821-00-0265']

We can thus deduce that all noisy data starts with "#"

Replacing values of noisy data

In [27]:
for index in range(len(df_ssn)):
    if df_ssn['SSN'][index].startswith('#'):
        df_ssn['SSN'][index] = [val for val in corresponding_unique_ssn_values[df_ssn['Customer_ID'][index]] if not val.startswith('#')][0]

In [28]:
df_ssn.head()

Customer_ID     Month           Name   Age          SSN Occupation  \
0   CUS_0xd40   January  Aaron Maashoh    23  821-00-0265  Scientist   
1   CUS_0xd40  February  Aaron Maashoh    23  821-00-0265  Scientist   
2   CUS_0xd40     March  Aaron Maashoh  -500  821-00-0265  Scientist   
3   CUS_0xd40     April  Aaron Maashoh    23  821-00-0265  Scientist   
4   CUS_0xd40       May  Aaron Maashoh    23  821-00-0265  Scientist   

  Annual_Income  Monthly_Inhand_Salary  Num_Bank_Accounts  Num_Credit_Card  \
0      19114.12            1824.843333                  3                4   
1      19114.12                    NaN                  3                4   
2      19114.12                    NaN                  3                4   
3      19114.12                    NaN                  3                4   
4      19114.12            1824.843333                  3                4   

   ...  Credit_Mix Outstanding_Debt Credit_Utilization_Ratio  \
0  ...           _           809.98                26.822620   
1  ...        Good           809.98                31.944960   
2  ...        Good           809.98                28.609352   
3  ...        Good           809.98                31.377862   
4  ...        Good           809.98                24.797347   

      Credit_History_Age Payment_of_Min_Amount Total_EMI_per_month  \
0  22 Years and 1 Months                    No           49.574949   
1                    NaN                    No           49.574949   
2  22 Years and 3 Months                    No           49.574949   
3  22 Years and 4 Months                    No           49.574949   
4  22 Years and 5 Months                    No           49.574949   

   Amount_invested_monthly                 Payment_Behaviour  \
0        80.41529543900253   High_spent_Small_value_payments   
1       118.28022162236736    Low_spent_Large_value_payments   
2          81.699521264648   Low_spent_Medium_value_payments   
3        199.4580743910713    Low_spent_Small_value_payments   
4       41.420153086217326  High_spent_Medium_value_payments   

      Monthly_Balance  Credit_Score  
0  312.49408867943663          Good  
1  284.62916249607184          Good  
2   331.2098628537912          Good  
3  223.45130972736786          Good  
4  341.48923103222177          Good  

[5 rows x 27 columns]

In [29]:
for id in unique_customer_id:
    corresponding_unique_ssn_values[id] = list(df_ssn[df_ssn['Customer_ID'] == id]['SSN'].unique())
dict(itertools.islice(corresponding_unique_ssn_values.items(), 5))

{'CUS_0xd40': ['821-00-0265'],
 'CUS_0x21b1': ['004-07-5839'],
 'CUS_0x2dbc': ['486-85-3974'],
 'CUS_0xb891': ['072-31-6145'],
 'CUS_0x1cdb': ['615-06-7821']}

making SSN more Readable by replacing - with empty value and making it a int value

In [30]:
for index in range(len(df_ssn)):
    df_ssn['SSN'][index] = int (df_ssn['SSN'][index].replace('-',''))
    

In [47]:
df_ssn['SSN'].nunique()

12500

Now dropping column Customer_ID because each row can now be identified by ssn

In [33]:
df_ssn.drop(columns = ['Customer_ID'],axis = 1, inplace = True)

In [36]:
df_ssn.to_csv('checkpoints//df_ssn.csv',index = False )

### PreProcessing Month

In [104]:
df_month = pd.read_csv('checkpoints//df_ssn.csv')

In [105]:
df_month['Month'].value_counts()

January     12500
February    12500
March       12500
April       12500
May         12500
June        12500
July        12500
August      12500
Name: Month, dtype: int64

Leaving months as it is for now

### PreProcessing Name

In [209]:
df_name = df_month.copy()

In [210]:
df_name['Name'].nunique()

10139

name maynot be that relevant so dropping

In [211]:
df_name.drop(columns = ['Name'], axis = 1, inplace = True)

In [212]:
df_name.head()

Month   Age        SSN Occupation Annual_Income  Monthly_Inhand_Salary  \
0   January    23  821000265  Scientist      19114.12            1824.843333   
1  February    23  821000265  Scientist      19114.12                    NaN   
2     March  -500  821000265  Scientist      19114.12                    NaN   
3     April    23  821000265  Scientist      19114.12                    NaN   
4       May    23  821000265  Scientist      19114.12            1824.843333   

   Num_Bank_Accounts  Num_Credit_Card  Interest_Rate Num_of_Loan  ...  \
0                  3                4              3           4  ...   
1                  3                4              3           4  ...   
2                  3                4              3           4  ...   
3                  3                4              3           4  ...   
4                  3                4              3           4  ...   

  Credit_Mix  Outstanding_Debt Credit_Utilization_Ratio  \
0          _            809.98                26.822620   
1       Good            809.98                31.944960   
2       Good            809.98                28.609352   
3       Good            809.98                31.377862   
4       Good            809.98                24.797347   

      Credit_History_Age  Payment_of_Min_Amount Total_EMI_per_month  \
0  22 Years and 1 Months                     No           49.574949   
1                    NaN                     No           49.574949   
2  22 Years and 3 Months                     No           49.574949   
3  22 Years and 4 Months                     No           49.574949   
4  22 Years and 5 Months                     No           49.574949   

  Amount_invested_monthly                 Payment_Behaviour  \
0       80.41529543900253   High_spent_Small_value_payments   
1      118.28022162236736    Low_spent_Large_value_payments   
2         81.699521264648   Low_spent_Medium_value_payments   
3       199.4580743910713    Low_spent_Small_value_payments   
4      41.420153086217326  High_spent_Medium_value_payments   

      Monthly_Balance Credit_Score  
0  312.49408867943663         Good  
1  284.62916249607184         Good  
2   331.2098628537912         Good  
3  223.45130972736786         Good  
4  341.48923103222177         Good  

[5 rows x 25 columns]

### PreProcessing Age

In [243]:
df_age = df_name.copy()

In [244]:
df_age['Age'].isna().sum()

0

Count possible noisy data in Age (age upper limit = 120 and lower limit is 0)

In [245]:
underscore_age = 0
underscore_age_list = []
noisy_age = 0
noisy_age_list = []
for index in range(len(df_age)):
    if '_' in  df_age['Age'][index]:
        underscore_age += 1
        underscore_age_list.append(df_age['Age'][index])
    try:
        if int(df_age['Age'][index]) > 120 or int(df_age['Age'][index] <= 0):
            noisy_age +=1
            noisy_age_list.append(df_age['Age'][index])
            
    except:
        pass
print("Underscore_age Count : ",underscore_age,"|| Unique Underscore_age count :",len(set(underscore_age_list)), " || Example : ",underscore_age_list[0:5])
print("Noisy_age Count : ",noisy_age,"|| Unique Noisy_age count :",len(set(noisy_age_list))," || Example : ",noisy_age_list[0:5])



Underscore_age Count :  4939 || Unique Underscore_age count : 127  || Example :  ['28_', '34_', '30_', '24_', '33_']
Noisy_age Count :  1799 || Unique Noisy_age count : 1608  || Example :  ['7580', '181', '995', '5079', '6409']


In [246]:
df_age['Age'].nunique()

1788

Probable Unique age values <br>
1788 - 127 - 1608 =53

Ommiting the underscore ages

In [247]:
for index in range(len(df_age)):
    if '_' in df_age['Age'][index]:
        df_age['Age'][index] = df_age['Age'][index].replace('_','')

Ommiting noisy ages

In [248]:
corresponding_unique_age_values= {}
unique_SSN = list(df_age['SSN'].unique())

In [249]:
df_age['Age'] = df_age['Age'].astype(int)

In [250]:
for ssn in unique_SSN:
    corresponding_unique_age_values[ssn] = list(df_age[df_age['SSN'] == ssn]['Age'].unique())

In [251]:
dict(itertools.islice(corresponding_unique_age_values.items(),10))

{821000265: [23, -500],
 4075839: [28],
 486853974: [34],
 72316145: [54, 55],
 615067821: [21],
 612708987: [31],
 411510676: [33, 34],
 500926408: [7580, 30],
 70191622: [23, 24],
 366681681: [44, 45]}

For now allowing multiple valid ages but replacing the noisy data

In [252]:
[val for val in corresponding_unique_age_values[df_age['SSN'][455]] if ((int(val) >= 0) and (int(val) <= 120)) ][0]

25

In [253]:
for index in range(len(df_age)):
    if df_age['Age'][index] > 120 or df_age['Age'][index] <= 0:
        df_age['Age'][index] = [val for val in corresponding_unique_age_values[df_age['SSN'][index]] if ((val > 0) and (val <= 120)) ][0]

Lets Check

In [254]:
noisy_age = 0
noisy_age_list = []
for index in range(len(df_age)):
    if df_age['Age'][index] > 120 or df_age['Age'][index] <= 0:
        noisy_age +=1
        noisy_age_list.append(df_age['Age'][index])
print("Noisy_age Count : ",noisy_age,"|| Unique Noisy_age count :",len(set(noisy_age_list))," || Example : ",noisy_age_list[0:5])

Noisy_age Count :  0 || Unique Noisy_age count : 0  || Example :  []


In [290]:
df_age['Age'].value_counts()

38     3076
28     3057
31     3043
26     3035
32     2968
36     2955
25     2952
35     2937
39     2932
27     2929
34     2922
44     2895
22     2876
19     2874
41     2863
20     2834
37     2824
29     2819
43     2814
21     2801
30     2799
24     2786
23     2725
45     2711
40     2687
42     2648
33     2624
18     2446
46     1663
15     1611
17     1551
16     1502
48     1427
49     1413
55     1396
53     1391
52     1391
54     1343
51     1319
50     1314
47     1250
14     1223
56      363
95        3
112       1
118       1
99        1
100       1
102       1
111       1
109       1
115       1
Name: Age, dtype: int64

In [257]:
df_age.to_csv('checkpoints//df_age.csv',index = False)

### PreProcessing Occupation

In [279]:
df_occupation = pd.read_csv('checkpoints//df_age.csv')

In [280]:
df_occupation.head()

Month  Age        SSN Occupation Annual_Income  Monthly_Inhand_Salary  \
0   January   23  821000265  Scientist      19114.12            1824.843333   
1  February   23  821000265  Scientist      19114.12                    NaN   
2     March   23  821000265  Scientist      19114.12                    NaN   
3     April   23  821000265  Scientist      19114.12                    NaN   
4       May   23  821000265  Scientist      19114.12            1824.843333   

   Num_Bank_Accounts  Num_Credit_Card  Interest_Rate Num_of_Loan  ...  \
0                  3                4              3           4  ...   
1                  3                4              3           4  ...   
2                  3                4              3           4  ...   
3                  3                4              3           4  ...   
4                  3                4              3           4  ...   

  Credit_Mix  Outstanding_Debt Credit_Utilization_Ratio  \
0          _            809.98                26.822620   
1       Good            809.98                31.944960   
2       Good            809.98                28.609352   
3       Good            809.98                31.377862   
4       Good            809.98                24.797347   

      Credit_History_Age  Payment_of_Min_Amount Total_EMI_per_month  \
0  22 Years and 1 Months                     No           49.574949   
1                    NaN                     No           49.574949   
2  22 Years and 3 Months                     No           49.574949   
3  22 Years and 4 Months                     No           49.574949   
4  22 Years and 5 Months                     No           49.574949   

  Amount_invested_monthly                 Payment_Behaviour  \
0       80.41529543900253   High_spent_Small_value_payments   
1      118.28022162236736    Low_spent_Large_value_payments   
2         81.699521264648   Low_spent_Medium_value_payments   
3       199.4580743910713    Low_spent_Small_value_payments   
4      41.420153086217326  High_spent_Medium_value_payments   

      Monthly_Balance Credit_Score  
0  312.49408867943663         Good  
1  284.62916249607184         Good  
2   331.2098628537912         Good  
3  223.45130972736786         Good  
4  341.48923103222177         Good  

[5 rows x 25 columns]

In [281]:
df_occupation["Occupation"].nunique()

16

In [282]:
df_occupation["Occupation"].value_counts()

_______          7062
Lawyer           6575
Architect        6355
Engineer         6350
Scientist        6299
Mechanic         6291
Accountant       6271
Developer        6235
Media_Manager    6232
Teacher          6215
Entrepreneur     6174
Doctor           6087
Journalist       6085
Manager          5973
Musician         5911
Writer           5885
Name: Occupation, dtype: int64

_______ seems to be noisy datas

In [283]:
corresponding_unique_occupation_values = {}

In [284]:
for ssn in unique_SSN:
    corresponding_unique_occupation_values[ssn] = list(df_occupation[df_occupation['SSN'] == ssn]['Occupation'].unique())

In [285]:
dict(itertools.islice(corresponding_unique_occupation_values.items(),10))

{821000265: ['Scientist'],
 4075839: ['_______', 'Teacher'],
 486853974: ['_______', 'Engineer'],
 72316145: ['Entrepreneur', '_______'],
 615067821: ['Developer'],
 612708987: ['Lawyer', '_______'],
 411510676: ['Lawyer'],
 500926408: ['Media_Manager', '_______'],
 70191622: ['Doctor'],
 366681681: ['Journalist']}

In [286]:
[val for val in corresponding_unique_occupation_values[df_occupation['SSN'][1]] if not val.startswith('_') ][0]

'Scientist'

In [287]:
for index in range(len(df_occupation)):
    if df_occupation['Occupation'][index].startswith('_'):
        df_occupation['Occupation'][index] = [val for val in corresponding_unique_occupation_values[df_occupation['SSN'][index]] if not val.startswith('_') ][0]

In [288]:
df_occupation["Occupation"].value_counts()

Lawyer           7096
Engineer         6864
Architect        6824
Mechanic         6776
Scientist        6744
Accountant       6744
Developer        6720
Media_Manager    6720
Teacher          6672
Entrepreneur     6648
Doctor           6568
Journalist       6536
Manager          6432
Musician         6352
Writer           6304
Name: Occupation, dtype: int64

In [289]:
for ssn in unique_SSN:
    corresponding_unique_occupation_values[ssn] = list(df_occupation[df_occupation['SSN'] == ssn]['Occupation'].unique())
dict(itertools.islice(corresponding_unique_occupation_values.items(),10))

{821000265: ['Scientist'],
 4075839: ['Teacher'],
 486853974: ['Engineer'],
 72316145: ['Entrepreneur'],
 615067821: ['Developer'],
 612708987: ['Lawyer'],
 411510676: ['Lawyer'],
 500926408: ['Media_Manager'],
 70191622: ['Doctor'],
 366681681: ['Journalist']}

### PreProcessing Annual Income

In [300]:
df_annual_income = df_occupation.copy()

In [301]:
df_annual_income["Annual_Income"].nunique()

18940

In [302]:
underscore_annual_income = 0
underscore_annual_income_list = []
for index in range(len(df_annual_income)):
    if '_' in  df_annual_income['Annual_Income'][index]:
        underscore_annual_income += 1
        underscore_annual_income_list.append(df_annual_income['Annual_Income'][index])
print("Underscore_annual_income Count : ",underscore_annual_income,"|| Unique Underscore_annual_income count :",len(set(underscore_annual_income_list)), " || Example : ",underscore_annual_income_list[0:5])

Underscore_annual_income Count :  6980 || Unique Underscore_annual_income count : 5503  || Example :  ['34847.84_', '30689.89_', '35547.71_', '34081.38_', '114838.41_']


Ommitting the underscore annual income

In [303]:
for index in range(len(df_annual_income)):
    if '_' in df_annual_income['Annual_Income'][index]:
        df_annual_income['Annual_Income'][index] = df_annual_income['Annual_Income'][index].replace('_','')

In [304]:
underscore_annual_income = 0
underscore_annual_income_list = []
for index in range(len(df_annual_income)):
    if '_' in  df_annual_income['Annual_Income'][index]:
        underscore_annual_income += 1
        underscore_annual_income_list.append(df_annual_income['Annual_Income'][index])
print("Underscore_annual_income Count : ",underscore_annual_income,"|| Unique Underscore_annual_income count :",len(set(underscore_annual_income_list)), " || Example : ",underscore_annual_income_list[0:5])

Underscore_annual_income Count :  0 || Unique Underscore_annual_income count : 0  || Example :  []


In [298]:
df_annual_income['Annual_Income'] = df_annual_income['Annual_Income'].astype(float)

In [305]:
df_annual_income.to_csv('checkpoints//df_annual_income.csv',index = False)

### PreProcessing Monthly_Inhand_Salary

In [306]:
df_monthly_inhand_salary = pd.read_csv('checkpoints//df_annual_income.csv')

In [307]:
df_monthly_inhand_salary['Monthly_Inhand_Salary'].nunique()

13235

In [308]:
df_monthly_inhand_salary['Monthly_Inhand_Salary'].isna().sum()

15002

### PreProcessing Num_Bank_Accounts

In [36]:
df_num_bank_accounts = df_monthly_inhand_salary.copy()

In [37]:
df_num_bank_accounts["Num_Bank_Accounts"].nunique()

943

### PreProcessing Num_Credit_Card

In [38]:
df_num_credit_card = df_num_bank_accounts.copy()

In [39]:
df_num_credit_card["Num_Credit_Card"].nunique()

1179

### PreProcessing Interest_Rate

In [40]:
df_interest_rate = df_num_credit_card.copy()

In [41]:
df_interest_rate["Interest_Rate"].nunique()

1750

### PreProcessing Num_of_Loan

In [42]:
df_num_of_loan = df_interest_rate.copy()

In [43]:
df_num_of_loan["Num_of_Loan"].nunique()

434

### PreProcessing Type_of_Loan

In [44]:
df_type_of_loan = df_num_of_loan.copy()

In [45]:
df_type_of_loan["Type_of_Loan"].nunique()

6260

### PreProcessing Delay_from_due_date

In [46]:
df_delay_from_due_date = df_type_of_loan.copy()

In [47]:
df_delay_from_due_date["Delay_from_due_date"].nunique()

73

### PreProcessing Num_of_Delayed_Payment

In [48]:
df_num_of_delayed_payment = df_delay_from_due_date.copy()

In [49]:
df_num_of_delayed_payment["Num_of_Delayed_Payment"].nunique()

749

### PreProcessing Changed_Credit_Limit

In [50]:
df_changed_credit_limit = df_num_of_delayed_payment.copy()

In [51]:
df_changed_credit_limit["Changed_Credit_Limit"].nunique()

4384

### PreProcessing Num_Credit_Inquiries

In [52]:
df_num_credit_inquires = df_changed_credit_limit.copy()

In [53]:
df_num_credit_inquires["Num_Credit_Inquiries"].nunique()

1223

### PreProcessing Credit_Mix

In [54]:
df_credit_mix = df_num_credit_inquires.copy()

In [55]:
df_credit_mix["Credit_Mix"].nunique()

4

### PreProcessing Outstanding_Debt

In [56]:
df_outstanding_debt = df_credit_mix.copy()

In [57]:
df_outstanding_debt["Outstanding_Debt"].nunique()

13178

### PreProcessing Credit_Utilization_Ratio

In [58]:
df_credit_utilization_ratio = df_outstanding_debt.copy()

In [59]:
df_credit_utilization_ratio["Credit_Utilization_Ratio"].nunique()

100000

### PreProcessing Credit_History_Age

In [60]:
df_credit_history_age = df_credit_utilization_ratio.copy()

In [61]:
df_credit_history_age["Credit_History_Age"].nunique()

404

### PreProcessing Payment_of_Min_Amount

In [62]:
df_payment_of_min_amount = df_credit_history_age.copy()

In [63]:
df_payment_of_min_amount["Payment_of_Min_Amount"].nunique()

3

### PreProcessing Total_EMI_per_month

In [64]:
df_total_emi_per_month = df_payment_of_min_amount.copy()

In [65]:
df_total_emi_per_month["Total_EMI_per_month"].nunique()

14950

### PreProcessing Amount_invested_monthly

In [66]:
df_amount_invested_monthly = df_total_emi_per_month.copy()

In [67]:
df_amount_invested_monthly["Amount_invested_monthly"].nunique()

91049

### PreProcessing Payment_Behaviour

In [69]:
df_payment_behaviour = df_amount_invested_monthly.copy()

In [70]:
df_payment_behaviour["Payment_Behaviour"].nunique()

7

### PreProcessing Monthly_Balance

In [71]:
df_monthly_balance = df_payment_behaviour.copy()

In [72]:
df_payment_behaviour["Payment_Behaviour"].nunique()

7

### PreProcessing Credit_Score

In [73]:
df_credit_score = df_payment_behaviour.copy()

In [74]:
df_credit_score["Credit_Score"].nunique()

3